### DISCUSSION TITLE
API Rules - Protect creating record with relations that doesn't belong to the user

### DISCUSSION THREAD

**Discussion by Penkie** [pocketbase/pocketbase#5667] [Q&A] [answered]
Hi,

I have two tables: **transactions** and **categories**. A transaction have a category linked to it, and both belong to the user. Nobody can see, edit, update or delete a transaction/category other the user linked to them.

See the schema:

**Transaction**
amount: number
user: User
category: Category

**Category**
name: string
user: User

I'm having trouble to protect my tables. Right now, the user can create a transaction with a categoryId that is not his. How can I protect this ?
I tried this on the create/update rule of the transaction table : @request.data.user = @request.auth.id && @collection.categories.user = @request.auth.id, but it doesn't seem to work as I have a 404.

Is there any solution to this ?

Thanks in advance !

**Comment by ganigeorgiev** [comment]
I want to understand from where the confusion come from and why people often reach unnecessary to the `@collection.*` field.

What parts of the API rules confuse you? Could you elaborate more on your reasoning and how did you end up with your rule?

---

To answer your question, you can use dot-notation to target the specific user. Your data structure seem denormalized (do you need the user to be defined in both places?) but one possible example create rule could be:

```js
@request.auth.id != "" &&
@request.data.user = @request.auth.id &&
@request.data.category.user = @request.auth.id
```

**Reply by Penkie** [reply, marked as answer]
Hi Gani, thanks for your reply. Just before I answer your question, just let me tell you that I really appreciate your work, and I am very thankfull for having such a great software to work with my project(s), so thank you !

I'm actually quite confused about the documentation ([https://pocketbase.io/docs/api-rules-and-filters/#api-rules](https://pocketbase.io/docs/api-rules-and-filters/#api-rules)), when reading this text under the @collection section: "This filter could be used to target other collections that are not directly related to the current one (aka. there is no relation field pointing to it) but both shares a common field value, like for example a category id:", I understand it as, okey so this will let me interact with other relations, and that's why I went for this "solution". But what is more confusing about it, is that the example given bellow is not really explained, and thus I tested this a found out it didn't work, and seeing nothing else in the documentation to help me, got here and asked.

About your answer, it does make sense. I felt before that I had to "search" or "get" the current category (as in @collection). It wasn't clear to me that you could access the whole record via @request because what I sent is just an id. But the way it works is perfect ! Just had to have someone tell me, maybe add a part in the documentation talking about this system ? As in system I mean "even if you give only an id in your body, in pocketbase the whole data is available if you want to validate some things". Just a humble suggestion, but maybe it's just me that doesn't understand this.

If I wasn't clear enough, don't hesitate to ask me, I'll be happy to answer.
